Test fixes for collections redesign

kdoh opened the pull request:
Updates to reflect changes in #13684

kdoh committed:
update dashboard create path in tests

kdoh committed:
merge 0.38-collections-redesign

kdoh committed:
update browse data clicks

kdoh committed:
Merge branch '0.38-collections-redesign' into 0.38-collections-test-fixes

nemanjaglumac committed:
Update collections Cypress tests

nemanjaglumac committed:
Remove `.only()` from a first `describe` block

nemanjaglumac commented on frontend/test/metabase/scenarios/collection_defaults.cy.spec.js:
```diff
@@ -12,80 +12,65 @@ const sub_collection_name = "ZZ Sub-Collection";
 
       it("should be the parent collection", () => {
         // Check that it has no parent
         const length = 7;
```
As a general rule/note: I don't like the idea of hard coding IDs. If we change the state of test DB, this could change and break.
Once everything else is done for this test suite, I'd like to address this issue and try to come up with a better solution.

kdoh committed:
Merge branch '0.38-collections-redesign' into 0.38-collections-test-fixes [ci skip]

nemanjaglumac committed:
Update `collection_defaults.cy.spec.js`

- Refactor hard coded ID references
- Add an additional check for a sub-collection within admin's personal collection

nemanjaglumac committed:
Tidy up comments

nemanjaglumac committed:
Fix broken `parameters.cy.spec.js`

nemanjaglumac commented:
At this point, there is only [one more test failing](https://app.circleci.com/pipelines/github/metabase/metabase/10275/workflows/3446e526-2b49-46db-9c33-cfedfe1ece21/jobs/364302) due to a temporary "bug?" in front-end code. The test itself doesn't need fixing. It lives in a `frontend/test/metabase/scenarios/permissions.cy.spec.js`

- **scenarios > permissions should display the permissions screen on /collection/1 - should display the permissions screen on /collection/1**
``` 
CypressError: Timed out retrying: Expected to find element: '.Icon-key', but never found it.
```

It should display a screen with keys icon and a message: "Sorry, you don’t have permission to see that."
`frontend/src/metabase/containers/ErrorPages.jsx:53`

but instead it seems to be rendering `frontend/src/metabase/components/LoadingAndErrorWrapper.jsx` with the error message coming from backend "You don't have permissions to do that."

nemanjaglumac committed:
Merge branch '0.38-collections-redesign' into 0.38-collections-test-fixes

nemanjaglumac committed:
Fix `admin_setup` smoke test [ci skip]

nemanjaglumac committed:
Isolate `.Icon-table_spaced` and extract it into a helper function

nemanjaglumac committed:
Fix issue in test with opening the correct dropdown

kdoh committed:
Merge branch '0.38-collections-redesign' into 0.38-collections-test-fixes
